### DISCUSSION TITLE
Leash Biosciences BELKA ligand prediction competition

### DISCUSSION THREAD

**Discussion by ian-quigley** [polaris-hub/polaris-recipes#26] [Datasets]
### Polaris Link

https://polarishub.io/datasets/leash-bio/belka-v1

### README

# The BELKA competition
This is the dataset provided for the [Big Encoded Library for Chemical Assessment (BELKA) competition](https://www.kaggle.com/competitions/leash-BELKA), which Leash Biosciences hosted on Kaggle in the summer of 2024. It is roughly 100M small molecules from DNA-encoded chemical libraries screened against 3 protein targets (BRD4, EPHX2/sEH, and ALB/HSA). More detailed descriptions of the targets, including links to target crystal structures for structure-based predictions, can be found [here](https://www.kaggle.com/competitions/leash-BELKA/data). 

### DELs
The vast majority of chemical matter in this dataset is a triazine-based 3-cycle library designed to superficially resemble DEL-A in ([1](https://pubmed.ncbi.nlm.nih.gov/19648931/)) and this library is called _AMA014_. AMA014 was designed and manufactured by [AlphaMa](https://www.alphama.com.cn/en/technologies/).

The orthogonal DEL in the library split is a non-triazine 3-cycle library designed by Leash to resemble kinase inhibitor chemistry and is called _kinase0_, or _kin0_ for short. 

## To benchmark against the BELKA competition
For the contest, Leash provided binary bind-or-no-bind labels for each small molecule pair. Leash also provided data splits: 1) a random molecule split, 2) a split where a central core (a triazine) is preserved but there are no shared building blocks between train and test, and 3) a library split. The library split is a series of screens with the same targets against an entirely different DEL with different building blocks, different cores, and different attachment chemistries. The library split was included in the test set only, to test true out-of-distribution predictions. 

This collection is identical to the datasets in the Kaggle competition in order to enable those who want to benchmark methods against it. The BELKA benchmark set can be found in the `test`,`train`, and `val` folders. 

### `Test-train-val`
Datasets in these folders represent the test/train/validation splits provided in the BELKA competition. They include binary labels from aggregated experimental rounds of selection and replicates. To facilitate exploration of a regression task, we included the numerical `molecule_score`.

### Columns in the BELKA competition dataset, in `test-train-val`
`molecule_smiles`: The structure of the fully assembled molecule, in SMILES. This includes the three building blocks and the triazine core if present. Note we use a `[Dy]` as the stand-in for the DNA linker.
`buildingblock1_smiles`, `buildingblock2_smiles`, `buildingblock3_smiles`: SMILES representation of the individual building blocks that, when combined with SMARTS reactions, produce the full molecule (also with a triazine core, in the case of AMA014) 
`protein_name`: Name of the target screened
`molecule_score`: a proprietary score incorporating enrichment of counts per `molecule_id` over a background condition
`binds`: a binary label of bind or no-bind derived from a cutoff applied to the molecule_score

## To try new methods of hit-calling and more
Maybe you want to know how replicates help, or if multiple rounds of selection is a good idea with AMA014, or what happens when you try different background conditions. This dataset is for you!

To allow exploration of additional hit-calling methods and the contributions of various aspects of experimental design to enriching those hits, Leash has also provided the raw/normalized sequencing counts and experimental details of each replicate. These can be found in folders following the sample label naming convention provided below, one folder per selection round/replicate.

#### Rounds of selection
A common DEL screening practice is to combine the library with the protein target, isolate the DEL/target complexes, elute the bound DEL (here, we did it with heat), and repeat the selection with fresh target protein (image below). The reasoning is that extra rounds select for only the tightest binders, making the most efficient use of sequencing capacity ([1](https://pubmed.ncbi.nlm.nih.gov/19648931/)). To test this method, Leash performed 3 rounds of selection on the triazine library per target protein and sequenced a small portion of each round. Kinase0 is a far smaller library, and only a single round of selection was performed on it.

![DEL selection schematic (2)](https://github.com/user-attachments/assets/61db51d8-3a57-42c8-9719-f25023fa45c4)

#### Number of replicates
Each series of selection rounds was performed in triplicate for AMA014. Kinase0 screens were performed in duplicate with a single round of selection.

#### Sample labels for AMA014
`[target]-[replicate]-[selection round]`
`HSA-A-R1/` is ALB/HSA, replicate 1, first selection round (see Rounds of selection above)
`SEH-C-R2/` is EPHX2/sEH, replicate 3, second selection round

#### Sample labels for kinase0
`[target]-[kin0]-[replicate]`
`SEH-Kin0-S1/` is EPHX2/sEH, kinase0, replicate 1, first (and only) selection round

#### Negative control conditions
One approach to identifying hits is by comparing the relative enrichment of sequenced barcodes over a background distribution. The distribution of the DEL by itself is such a background; an alternate background would be to mix the DEL and the magnetic beads used in selection to identify a bead-only distribution. Leash is providing both of these. 

#### Sample labels for DEL-only negative controls
Leash sequenced the DELs alone in duplicate to understand the background distribution of the libraries.
`DEL-A`, `DEL-B` are replicates of AMA014 alone.
`DEL-Kin0-S1`, `DEL-Kin0-S2` are replicates of kinase0 alone.

#### Sample labels for bead-only negative controls
To compare bead-only signal against the AMA014 selections, Leash used [His Dynabeads from Thermo](https://www.thermofisher.com/order/catalog/product/10103D?SID=srch-srp-10103D) in triplicate. The Dynabeads were used in the protein-target selections of AMA014.
`His-A`, `His-B`, and `His-C` are replicates of Dynabead-only selections, one selection round only.

To compare bead-only signal against the kinase0 selections, Leash used [EDTA-compatible Magbeads from Thermo](https://www.thermofisher.com/order/catalog/product/A50589) in duplicate. The Magbeads were used in the protein-target selections of kinase0. Some building blocks in kinase0 were strong Dynabead binders, which is why the Magbeads were used in these screens.
`HisEDTA-Kin0-S1` and `HisEDTA-Kin0-S2` are replicates of Magbead-only selections, one selection round only. 

### Columns in the raw dataset (individual experiment folders)
`molecule_id`: unique identifier per molecule
`smiles`: SMILES representation of the full molecule 
`bb1_smiles`, `bb2_smiles`, `bb3_smiles`: SMILES representation of the individual building blocks that, when combined using SMARTS reactions, produce the full molecule (also with a triazine core, in the case of AMA014) 
`bcj_id`: binding count job, an id created internally at Leash to track computational counting jobs
`raw_counts`: read counts obtained from the sequencing data
`counts_per_billion`: read counts normalized by the size of the entire sequencing run and scaled to one billion reads

Have fun!

### Dataset Source

Andrew Blevins, Ian K Quigley, Brayden J Halverson, Nate Wilkinson, Rebecca S Levin, Agastya Pulapaka, Walter Reade, and Addison Howard. NeurIPS 2024 - Predict New Medicines with BELKA. https://kaggle.com/competitions/leash-BELKA, 2024. Kaggle.

### Dataset Curation

All data was generated internally by Leash Biosciences, using the same reagents, facility, and talent. We propose a ML-ready dataset that aggregates raw readouts using a proprietary scoring method, but also provide all raw data so the community can try different hit-calling methods.

### Dataset Completeness

- [X] I confirm that I filled out at least the `readme`, `source` and `curation_reference` fields for my Polaris dataset.

**Comment by cwognum** [comment]
Hey @ian-quigley , thank you for the submission! 

This is an awesome resource you're sharing with the community!

Referring back to our [Dataset 101 Guidelines](https://polarishub.io/datasets-101), we are happy to certify this!
1. On relevance: DEL screens are a common technique to identify hits.
2. On the data consistency: The dataset clearly stems from a consistent source, since it was generated internally. 
3. On obvious errors / ambiguous data: I did a quick check and all looks good. The fact that this dataset was used for a Kaggle competition by 2000+ people already gives me extra confidence.

Curious to see if people can improve over the OOD performance that was observed in the Kaggle competition. Also interested to see how people will use the raw data you provided.

Thanks again for sharing! 🙏

**Comment by cwognum** [comment]
I would be curious to hear what @PatWalters thinks of this dataset as well!

**Reply by PatWalters** [reply]
I have an entire upcoming blog post dedicated to this dataset.
